src/cs/Ssh.Tcp/IPortForwardMessageFactory.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+using System.Threading.Tasks;
+using Microsoft.DevTunnels.Ssh.Messages;
+
+namespace Microsoft.DevTunnels.Ssh.Tcp;
+
+/// <summary>
+/// Enables applications to extend port-forwarding by providing custom message subclasses
+/// that may include additional properties.
+/// </summary>
+/// <remarks>
+/// Custom message subclasses must override <see cref="SshMessage.OnRead" /> and
+/// <see cref="SshMessage.OnWrite" /> to handle serialization of any additional properties.
+/// </remarks>
+public interface IPortForwardMessageFactory
+{
+	/// <summary>
+	/// Creates a message for requesting to forward a port.
+	/// </summary>
+	/// <returns>An instance or subclass of <see cref="PortForwardRequestMessage" />.</returns>
+	Task<PortForwardRequestMessage> CreateRequestMessageAsync(int port);
```

Comment:
Should it be `remotePort`?

Reply:
I'm not sure that name will be clear enough in this context. But I'll add doc-comments explaining exactly what that parameter refers to.

Reply:
Updated.